### PR TITLE
Adding support for aws s3 storage

### PR DESCRIPTION
**Pull request by erez-work**

### CODE REVIEW

File: src/storage.js [hunk 1]
```diff
@@ -114,9 +114,71 @@ const storageService = (provider, connectionString) => {
     };
   };
 
+  const makeAwsStorage = () => {
+    const AWS = require('aws-sdk');
+    const s3 = new AWS.S3();
+
+    const createWriteStream = (bucketName, fileName) => {
```
Comment (nitzo) [inline src/storage.js:121]: (bucketName, fileName) => throw new Error("createWriteStream is not supported for AWS")
(this function body {} is redundant)

File: src/storage.js [hunk 2]
```diff
@@ -114,9 +114,71 @@ const storageService = (provider, connectionString) => {
...
+      throw new Error("createWriteStream is not supported for AWS");
+    };
+
+    const getSignedUrl = (bucketName, fileName) => {
```
Comment (nitzo) [inline src/storage.js:125]: same here, function bode is redundant, also inside the promise and in the next functions

File: src/storage.js [hunk 3]
```diff
@@ -114,9 +114,71 @@ const storageService = (provider, connectionString) => {
...
+          Bucket: bucketName,
+          Key: fileName,
+          Expires: 60 * 5
+        }))
+      });
+    };
+
+    const downloadFileAsJson = (bucketName, fileName) => {
+      return new Promise((resolve, reject) => {
+        try {
```
Comment (nitzo) [inline src/storage.js:137]: the try catch is mispalced. needs to be inside the callback...

File: src/storage.js [hunk 4]
```diff
@@ -114,9 +114,71 @@ const storageService = (provider, connectionString) => {
...
+        } catch (e) {
+          reject("File is not in JSON format");
+        }
+      });
+    };
+
+
+    const uploadJsonBlob = (container, blobName, payload) => {
+      return new Promise((resolve, reject) => {
+        try {
```
Comment (nitzo) [inline src/storage.js:154]: save here, try/catch is misplaced

File: src/storage.js (hunk 1 above)
Comment (erez-work) [reply src/storage.js:121]: It's a syntax error without the `{}`

File: src/storage.js (hunk 4 above)
Comment (erez-work) [reply src/storage.js:154]: This one is correct. the parsing is done before the calling the API

**Pull request merged**